File: api/Core/Model/CustomerBankingDetails.cs
using System;

namespace Xcianify.Core.Model
{
    public class CustomerBankingDetails : BaseModel
    {
        public int CustomerId { get; set; }
        public string? BankIfscCode { get; set; }
        public string? BankAccountNumber { get; set; }
        public string? BankName { get; set; }
        public string? CustomerBanker { get; set; }
        public string? CustomerVpa { get; set; }
        public string? BankAccountTypeCode { get; set; }
        public string? BankBranch { get; set; }
        public string? BankLocation { get; set; }
        public bool IsPrimary { get; set; } = true;
    }
} 

File: pharma-erp-3/types/customer-banking-details.ts
export interface CustomerBankingDetails {
  id: number;
  customerId: number;
  bankIfscCode?: string;
  bankAccountNumber?: string;
  bankName?: string;
  customerBanker?: string;
  customerVpa?: string;
  bankAccountTypeCode?: string;
  bankBranch?: string;
  bankLocation?: string;
  isPrimary: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface CustomerBankingDetailsFilter {
  customerId: number;
  isPrimary?: boolean;
  page: number;
  pageSize: number;
  sortBy: string;
  sortDescending: boolean;
}

export type CreateCustomerBankingDetailsData = Omit<CustomerBankingDetails, 'id' | 'createdAt' | 'updatedAt'>;

export type UpdateCustomerBankingDetailsData = Partial<Omit<CustomerBankingDetails, 'id' | 'createdAt' | 'updatedAt'>> & {
  id: number;
};
